AsyncDisplayKit/ASCollectionView.mm

```diff
@@ -313,6 +332,17 @@ - (void)setAsyncDelegate:(id<ASCollectionViewDelegate>)asyncDelegate
     super.delegate = (id<UICollectionViewDelegate>)_proxyDelegate;
     _asyncDelegateImplementsInsetSection = ([_asyncDelegate respondsToSelector:@selector(collectionView:layout:insetForSectionAtIndex:)] ? 1 : 0);
   }
+
+  [_flowLayoutInspector cacheSelectorsForCollectionView:self];
+}
+
+- (void)setCollectionViewLayout:(UICollectionViewLayout *)collectionViewLayout
```

Comment:
Are you totally sure that this behavior will work?  Have you tested setting a different collection layout after the ASCollectionView is in full swing operation?  I'd say we shouldn't provide the method at all until it has been tested, for example switching the collection layout while the current collection is scrolling at high speed.  In this implementation I don't see a call to -reloadData or anything, and beyond that there probably needs to be an even more explicit "destroy all cached data" in the ASDataController or something...really not sure, just anxious :)